collective/recipe/plonesite/plonesite.py

```diff
@@ -71,17 +78,19 @@ def runProfiles(plone, profiles):
 
 
 def quickinstall(plone, products):
-    logger.info("Quick installing: %s", products)
-    qit = plone.portal_quickinstaller
-    not_installed_ids = [
-        x['id'] for x in qit.listInstallableProducts(skipInstalled=1)]
-    installed_ids = [x['id'] for x in qit.listInstalledProducts()]
-    installed_products = list(filter(installed_ids.count, products))
-    not_installed = list(filter(not_installed_ids.count, products))
-    if installed_products:
-        qit.reinstallProducts(installed_products)
-    if not_installed_ids:
-        qit.installProducts(not_installed)
+    if PLONE6:
+        logger.info("Deprecation notice: Install products via quickinstall is deprecated in Plone 6. Use profiles instead.")
```

Comment:
This is not a deprecation. If something is deprecated, then it still works but you get a warning that it will no longer work in the next version. Here it simply no longer works. And basically it silently fails: you get an info log and the code continues without doing anything.
I would say, log an error ("Installing products by name is no longer supported in Plone 6. Use profiles instead.") and raise a ValueError or some other exception. If someone uses this on Plone 6, they are doing it wrong.

Tehnically, you could use [`install_product`](https://github.com/plone/Products.CMFPlone/blob/5.2.3/Products/CMFPlone/controlpanel/browser/quickinstaller.py#L425):

```
from Products.CMFPlone.utils import get_installer
installer = get_installer(plone)
for product in products:
    installer.install_product(product)
```

This will look for the default install profile of the product.
But it requires yet another condition in the code: check if `get_installer` is available in this Plone version. You may or may not want that.

Reply:
You are correct, it's not a deprecation. I will amend that. If it's not working in Plone 6, and we are encouraging people to use profiles instead, IMHO it should raise the error. Maybe remove it completely in a major when 6 is out.